### PR TITLE
change common beginning to common path

### PR DESCRIPTION
**Pull request by tonny008**
### What does it do?

Change common beginning to common path

### Why is it needed?
server.js
url: http://example.com/api
admin.url http://example.com/admin

use common beginning the admin path will be `dmin`

### CODE REVIEW

**Comment by petersg83** [comment]
Hi :)
Thanks for you contribution. The code can be improved indeed.

I read your code, here is a proposition of something even shorter and generic:
```js
const getCommonBeginning = (...strings) => takeWhile(
  strings[0],
  (char, index) => strings.every(string => string[index] === char)
).join('');
```

Can you directly change the function `getCommonBeginning` in `/packages/core/utils/lib/string-formatting.js`?
Also can you add some tests for this function in `/packages/core/utils/lib/__tests__/string-formatting.test.js`

**Review by petersg83 — CHANGES_REQUESTED** [review]
Hi :)
Thanks for you contribution. The code can be improved indeed.

I read your code, here is a proposition of something even shorter and generic:
```js
const getCommonBeginning = (...strings) => takeWhile(
  strings[0],
  (char, index) => strings.every(string => string[index] === char)
).join('');
```

Can you directly change the function `getCommonBeginning` in `/packages/core/utils/lib/string-formatting.js`?
Also can you add some tests for this function in `/packages/core/utils/lib/__tests__/string-formatting.test.js`?

**Comment by tonny008** [comment]
I don’t think it’s common beginnings here.

Given:
http://example.com/api
http://example.com/admin

common beginnings is http://example.com/a,
Common path is http://example.com

use common beginning the admin path will be dmin

**Comment by petersg83** [comment]
You're right I didn't see that. Thank you.
Can you please move the function next to `commonBeginning` in `/packages/core/utils/lib/string-formatting.js`?
And if you have time to write a few tests in `/packages/core/utils/lib/__tests__/string-formatting.test.js`, it would be perfect.
Otherwise I can do it too

File: packages/strapi-utils/lib/__tests__/string-formatting.test.js [hunk 1]
```diff
@@ -64,4 +64,29 @@ describe('string-formatting', () => {
...
+      const result = getCommonBeginning(...a);
+      expect(result).toBe(expectedResult);
+    });
+  });
+
+  describe('getCommonPath', () => {
+    const tests = [
+      [['abc', 'ab'], ''],
+      [['http://ab.com/cd', 'http://ab.com/c'], 'http://ab.com'],
+      [['http://ab.com/admin', 'http://ab.com/api'], 'http://ab.com'],
```
Comment (petersg83) [inline packages/strapi-utils/lib/__tests__/string-formatting.test.js:84]: Can you add:
```js
[['http://ab.com/admin', 'http://ab.com/admin/'], 'http://ab.com/admin'],
[['http://ab.com/admin', 'http://ab.com/admin'], 'http://ab.com/admin'],
```

File: packages/strapi-utils/lib/__tests__/string-formatting.test.js [hunk 2]
```diff
@@ -64,4 +64,29 @@ describe('string-formatting', () => {
       expect(result).toBe(expectedResult);
     });
   });
+
+  describe('getCommonBeginning', () => {
+    const tests = [
+      [['abcd', 'abc', 'ab'], 'ab'],
+      [['abcd', 'abc'], 'abc'],
+      [['ab/cd', 'ab/c'], 'ab/c'],
```
Comment (petersg83) [inline packages/strapi-utils/lib/__tests__/string-formatting.test.js:72]: Can you add:
```js
[['abc', 'abc'], 'abc'],
```

**Review by petersg83 — COMMENTED** [review]
Thank you, you did everything :D 
I proposed a couple more tests. What do you think?

**Review by petersg83 — DISMISSED** [review]
It looks good to me! Thank you for this contribution!

File: packages/strapi-utils/lib/string-formatting.js [hunk 3]
```diff
@@ -1,23 +1,21 @@
...
-    }
-  }
-  return common;
+const getCommonBeginning = (...strings) => _.takeWhile(
+  strings[0],
+  (char, index) => strings.every(string => string[index] === char)
+).join('');
+
+const getCommonPath = (...paths)=> {
+  const [segments, ...otherSegments] = paths.map(it=>_.split(it, '/'));
```
Comment (alexandrebodin) [inline packages/strapi-utils/lib/string-formatting.js:15]: can you add some spacing to make the code more readable.

**Comment by alexandrebodin** [comment]
Hi there 👋  Do you think you will be able to finish the changes requested at some point ?

**Review by petersg83 — APPROVED** [review]

**Pull request merged**